### PR TITLE
Add method to get the request body

### PR DESCRIPTION
**Pull request by k-nasa**
## Description

close: https://github.com/rustasync/surf/issues/67

Currently there seems to be no method to get the request body.
So I added it.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by k-nasa** [comment]
How about this?

File: src/request.rs [hunk 1]
```diff
@@ -558,6 +626,11 @@ impl<C: HttpClient> Request<C> {
     pub fn request(&self) -> Option<&http_client::Request> {
         self.req.as_ref()
     }
+
+    /// Get a mutable HTTP request
+    pub fn request_mut(&mut self) -> Option<&mut http_client::Request> {
+        self.req.as_mut()
+    }
```
Comment (yoshuawuyts) [inline src/request.rs]: We probably don't want to expose this as part of our public API.

File: src/request.rs [hunk 2]
```diff
@@ -469,6 +470,73 @@ impl<C: HttpClient> Request<C> {
...
+    /// # Ok(()) }
+    /// ```
+    pub async fn read_body_to_json<T: DeserializeOwned>(&mut self) -> Result<T, Exception> {
+        let req = self.request_mut().unwrap();
+
+        let mut bytes = Vec::new();
+        req.body_mut().read_to_end(&mut bytes).await?;
+
+        Ok(serde_json::from_slice(&bytes)?)
+    }
```
Comment (yoshuawuyts) [inline src/request.rs]: isn't this the same as the `body_json` method?

**Comment by yoshuawuyts** [comment]
@k-nasa I don't think we can do this because we have to assume data cannot be read multiple times, which means "inspecting" data would be equal to consuming it. I've written in more detail about it here: https://github.com/http-rs/surf/issues/67#issuecomment-554434153.

**Pull request closed without merging**